examples/camera-app/linux/src/camera-device.cpp

```diff
@@ -37,6 +39,176 @@ using namespace chip::app::Clusters::CameraAvSettingsUserLevelManagement;
 
 using namespace Camera;
 
+namespace {
+
+// Function to unreference GStreamer elements and the pipeline
+template <typename... Args>
+void unrefGstElements(GstElement * pipeline, Args... elements)
+{
+    if (pipeline)
+    {
+        gst_object_unref(pipeline);
+    }
+
+    // Unreference each element in the variadic template argument pack
+    ((elements ? gst_object_unref(elements) : void()), ...);
+}
+
+} // namespace
+
+// Namespace for GStreamer snapshot pipeline creation and camera type detection
+namespace GstSnapshot {
+
+enum class CameraType
+{
+    CSI,
+    USB,
+    Other,
+    Unkown,
+};
+
+CameraType detectCameraType(const std::string & fullDevicePath)
+{
+    if (!std::filesystem::exists(fullDevicePath))
+    {
+        return CameraType::Unkown;
+    }
+
+    std::string videoDeviceName = std::filesystem::path(fullDevicePath).filename(); // ex: video0
+    std::string sysPath         = "/sys/class/video4linux/" + videoDeviceName + "/device/driver/module";
+
+    char resolvedPath[PATH_MAX];
+    ssize_t len = readlink(sysPath.c_str(), resolvedPath, sizeof(resolvedPath) - 1);
+
+    // Define driver name constants to avoid magic strings
+    constexpr const char * kCsiDriver1 = "bm2835";
+    constexpr const char * kCsiDriver2 = "unicam";
+    constexpr const char * kUsbDriver  = "uvc";
+
+    if (len != -1)
+    {
+        resolvedPath[len] = '\0';
+        std::string driverPath(resolvedPath);
+
+        if (driverPath.find(kCsiDriver1) != std::string::npos || driverPath.find(kCsiDriver2) != std::string::npos)
+        {
+            return CameraType::CSI;
+        }
+        else if (driverPath.find(kUsbDriver) != std::string::npos)
```

Comment:
nit: prefer no else after return. clang-tidy does not check examples, but we generally try to abide to https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html